File: Client/src/main/java/jk_5/nailed/client/network/ClientNetworkHandler.java
package jk_5.nailed.client.network;

import cpw.mods.fml.common.network.FMLEmbeddedChannel;
import cpw.mods.fml.common.network.FMLOutboundHandler;
import cpw.mods.fml.common.network.NetworkRegistry;
import cpw.mods.fml.relauncher.Side;
import io.netty.channel.ChannelPipeline;
import jk_5.nailed.client.network.handlers.*;
import jk_5.nailed.client.scripting.ScriptPacketHandler;
import jk_5.nailed.network.NailedPacket;
import jk_5.nailed.network.NailedPacketCodec;

/**
 * No description given
 *
 * @author jk-5
 */
public class ClientNetworkHandler {

    private static FMLEmbeddedChannel channel;

    public static void registerChannel(){
        channel = NetworkRegistry.INSTANCE.newChannel("nailed", new NailedPacketCodec()).get(Side.CLIENT);

        ChannelPipeline pipeline = channel.pipeline();
        String targetName = channel.findChannelHandlerNameForType(NailedPacketCodec.class);

        pipeline.addAfter(targetName, "NotificationHandler", new NotificationHandler());
        pipeline.addAfter(targetName, "OpenGuiHandler", new OpenGuiHandler());
        pipeline.addAfter(targetName, "TileEntityDataHandler", new TileEntityDataHandler());
        pipeline.addAfter(targetName, "TimeUpdateHandler", new TimeUpdateHandler());
        pipeline.addAfter(targetName, "SkinDataHandler", new SkinDataHandler());
        pipeline.addAfter(targetName, "StoreSkinHandler", new StoreSkinHandler());
        pipeline.addAfter(targetName, "MapDataHandler", new MapDataHandler());
        pipeline.addAfter(targetName, "ParticleHandler", new ParticleHandler());
        pipeline.addAfter(targetName, "TerminalGuiHandler", new TerminalGuiHandler());
        pipeline.addAfter(targetName, "MapEditHandler", new MapEditHandler());

        pipeline.addAfter(targetName, "Script-MachineUpdateHandler", new ScriptPacketHandler.MachineUpdateHandler());
    }

    public static void sendPacketToServer(NailedPacket packet){
        channel.attr(FMLOutboundHandler.FML_MESSAGETARGET).set(FMLOutboundHandler.OutboundTarget.TOSERVER);
        channel.writeOutbound(packet);
    }
}

File: Client/src/main/scala/jk_5/nailed/client/network/ClientNetworkHandler.scala
package jk_5.nailed.client.network

import cpw.mods.fml.common.network.{FMLOutboundHandler, NetworkRegistry, FMLEmbeddedChannel}
import jk_5.nailed.network.{NailedPacket, NailedPacketCodec}
import cpw.mods.fml.relauncher.Side

/**
 * No description given
 *
 * @author jk-5
 */
object ClientNetworkHandler {
  private var channel: FMLEmbeddedChannel = null

  def registerChannel(){
    this.channel = NetworkRegistry.INSTANCE.newChannel("nailed", new NailedPacketCodec).get(Side.CLIENT)

    val pipeline = channel.pipeline
    val targetName = channel.findChannelHandlerNameForType(classOf[NailedPacketCodec])
    pipeline.addAfter(targetName, "OpenGuiHandler", OpenGuiHandler)
    pipeline.addAfter(targetName, "TileEntityDataHandler", TileEntityDataHandler)
    pipeline.addAfter(targetName, "TimeUpdateHandler", TimeUpdateHandler)
    pipeline.addAfter(targetName, "ParticleHandler", ParticleHandler)
    pipeline.addAfter(targetName, "TerminalGuiHandler", TerminalGuiHandler)
    pipeline.addAfter(targetName, "MapEditHandler", MapEditHandler)
    pipeline.addAfter(targetName, "DisplayLoginHandler", DisplayLoginHandler)
    pipeline.addAfter(targetName, "LoginResponseHandler", LoginResponseHandler)
    pipeline.addAfter(targetName, "FieldStatusHandler", FieldStatusHandler)
    pipeline.addAfter(targetName, "MachineUpdateHandler", MachineUpdateHandler)
    pipeline.addAfter(targetName, "TeamInformationHandler", TeamInformationHandler)
  }

  def sendPacketToServer(packet: NailedPacket){
    channel.attr(FMLOutboundHandler.FML_MESSAGETARGET).set(FMLOutboundHandler.OutboundTarget.TOSERVER)
    channel.writeOutbound(packet)
  }
}
